transcript/src/main/java/com/transcriptanalyzer/transcript/User_Requests_Intents/Service/TranscriptService.java

```diff
@@ -112,48 +130,101 @@ private static void flowIterator2(ArrayList<ArrayList<ArrayList<String>>> finalP
-//            System.out.println(turnContent);
 
+//          If the turn is identified as one with user input ("query"), isolate the actual intent ("name") as a string
+//          and return it contained in an ArrayList.
             if (turnContent.keySet().contains("query") || turnContent.keySet().contains("intent")) {
-                String intent = turnContent.get("intent").toString();
-                contains.add("intent: " + intent);
+                JsonObject intentContainer = turnContent.get("intent").getAsJsonObject();
+                String intent = intentContainer.get("name").toString();
+                contains.add(intent);
             }
 
+//          If the turn is identified to be a chatbot response ("message"), return the message as a string within an
+//          ArrayList.
             if (turnContent.keySet().contains("message")){
                 String message = turnContent.get("message").toString();
-                contains.add("message: " + message);
+                contains.add(message);
             }
-
+//          Store result of the parsed turn
             overallParse.add(contains);
         }
+
+//      Results of all turns being parsed.
         finalParseResults.add(overallParse);
     }
 
+    private static ArrayList<ArrayList<ArrayList<String>>> finalMerge(ArrayList<ArrayList<ArrayList<String>>> finalParse) {
+
+//      Final return, see meaning of each layer in getJsonContent above.
+        ArrayList<ArrayList<ArrayList<String>>> mergedReturn = new ArrayList<>();
+
+//      Access each transcript individually.
+        for(ArrayList<ArrayList<String>> overallParse : finalParse) {
+            int intentIndex = 1;
+            ArrayList<ArrayList<String>> mergedResult = new ArrayList<>();
+
+            while (intentIndex <= overallParse.size()) {
+
+
+                ArrayList<String> turnToMerge = new ArrayList<>();
+                StringBuilder currMessage = new StringBuilder();
+                StringBuilder currIntent = new StringBuilder();
+
+//              Check if we are at the end of a successful transcript
+                if (intentIndex == overallParse.size()) {
+                    currMessage.append(overallParse.get(intentIndex - 1).get(0));
+                    currIntent.append("Null");
+                }
+                else {
+                    currMessage.append(overallParse.get(intentIndex - 1).get(0));
+                    currIntent.append(overallParse.get(intentIndex).get(0));
+                }
+
+                turnToMerge.add(String.valueOf(currMessage));
+
+//              Isolate messages of end turns of successful transcripts
+                if (!String.valueOf(currIntent).equals("Null")) {
+                    turnToMerge.add(String.valueOf(currIntent));
+                }
+
+                mergedResult.add(turnToMerge);
+                intentIndex = intentIndex + 2;
```

Comment:
Minor change, but we may want to consider extracting this while loop into a helper method to reduce the bulk of this function.